### PR TITLE
[MSPAINT] Prepare for debugging

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
This PR helps the developers debug our Paint software.
JIRA issue: [CORE-17969](https://jira.reactos.org/browse/CORE-17969)

## Proposed changes

- Add `#include <debug.h>` and add link to `ntdll.dll` for debugging purpose.
- Use `DPRINT` for debugging.
- Add `CopyDIBImage` helper function for simplification.
- Refactor some codes.

## Screenshots
image
LGTM.

### CODE REVIEW

File: base/applications/mspaint/dib.cpp [hunk 1]
```diff
@@ -208,3 +208,8 @@ HBITMAP DoLoadImageFile(HWND hwnd, LPCTSTR name, BOOL fIsMainFile)
 
     return hBitmap;
 }
+
+HBITMAP CopyDIBImage(HBITMAP hbm, INT cx, INT cy)
```
Comment (HBelusca) [inline base/applications/mspaint/dib.cpp:212]: Could be made as an inlined function.

File: base/applications/mspaint/precomp.h [hunk 2]
```diff
@@ -21,6 +21,17 @@
...
+#define NDEBUG
+#include <reactos/debug.h>
+
+#if DBG
+    #define __LOCATION__  LPCSTR file, INT line
+    #define __location__  __FILE__, __LINE__
+#else
+    #define __LOCATION__  /*empty*/
+    #define __location__  /*empty*/
+#endif
```
Comment (HBelusca) [inline base/applications/mspaint/precomp.h:33]: This is quite an invasive debugging way! I would suggest to keep this PR not as something that goes into master, but as a "tool" to apply only when local debugging of Paint is needed, and that needs to be removed afterwards once debugging has reached its goals 🤔

File: base/applications/mspaint/dib.cpp (hunk 1 above)
Comment (katahiromz) [reply base/applications/mspaint/dib.cpp]: OK.

File: base/applications/mspaint/precomp.h (hunk 2 above)
Comment (katahiromz) [reply base/applications/mspaint/precomp.h]: `__LOCATION__` and `__location__` removed.

File: base/applications/mspaint/precomp.h [hunk 3]
```diff
@@ -21,6 +21,9 @@
 #include <shellapi.h>
 #include <htmlhelp.h>
 
+#define NDEBUG
+#include <reactos/debug.h>
```
Comment (HBelusca) [inline base/applications/mspaint/precomp.h:25]: `#include <debug.h>` should be sufficient? or would this break compilation? (if so, then just keep using `reactos/debug.h`)

File: base/applications/mspaint/history.cpp [hunk 4]
```diff
@@ -183,9 +186,9 @@ void ImageModel::StretchSkew(int nStretchPercentX, int nStretchPercentY, int nSk
 {
     int oldWidth = GetWidth();
     int oldHeight = GetHeight();
-    Insert((HBITMAP) CopyImage(hBms[currInd], IMAGE_BITMAP,
-           GetWidth() * nStretchPercentX / 100,
-           GetHeight() * nStretchPercentY / 100, 0));
+    INT newWidth = GetWidth() * nStretchPercentX / 100;
+    INT newHeight = GetHeight() * nStretchPercentY / 100;
```
Comment (HBelusca) [inline base/applications/mspaint/history.cpp:190]:
```suggestion
    INT newWidth = oldWidth * nStretchPercentX / 100;
    INT newHeight = oldHeight * nStretchPercentY / 100;
```

File: base/applications/mspaint/history.cpp [hunk 5]
```diff
@@ -44,8 +44,9 @@ ImageModel::ImageModel()
 
 void ImageModel::CopyPrevious()
 {
+    DPRINT("CopyPrevious: %d\n", currInd);
```
Comment (HBelusca) [inline base/applications/mspaint/history.cpp:47]: That's ok; also, you could use `__FUNCTION__` if you don't want to hardcode the name of the function.

File: base/applications/mspaint/precomp.h (hunk 3 above)
Comment (katahiromz) [reply base/applications/mspaint/precomp.h]: Used `<debug.h>`.

File: base/applications/mspaint/history.cpp (hunk 4 above)
Comment (katahiromz) [reply base/applications/mspaint/history.cpp]: OK.

File: base/applications/mspaint/history.cpp (hunk 5 above)
Comment (katahiromz) [reply base/applications/mspaint/history.cpp]: OK.

**Pull request merged**